frontend/lib/identification/activation_workflow/activation_code_scanner_page.dart

```diff
@@ -107,15 +109,21 @@ class ActivationCodeScannerPage extends StatelessWidget {
           throw const ActivationInvalidTotpSecretException();
         }
         final totpSecret = const Base64Decoder().convert(activationResult.totpSecret!);
-        debugPrint('Card Activation: Successfully activated.');
 
-        provider.setCode(DynamicUserCode()
+        DynamicUserCode userCode = DynamicUserCode()
           ..info = activationCode.info
           ..pepper = activationCode.pepper
           ..totpSecret = totpSecret
           ..cardVerification = (CardVerification()
             ..cardValid = true
-            ..verificationTimeStamp = secondsSinceEpoch(DateTime.parse(activationResult.activationTimeStamp))));
+            ..verificationTimeStamp = secondsSinceEpoch(DateTime.parse(activationResult.activationTimeStamp)));
+
+        userCodesModel.insertCode(userCode);
+        if (userCodesModel.userCodes.length > 1) {
+          moveToLastCard();
+        }
```

Comment:
```suggestion
        moveToLastCard();
```

Reply:
yes unfortunately the slider library causes an issue if there is only one item, thats why i added it